packages/arb-bridge-eth/contracts/rollup/Node.sol

```diff
@@ -17,131 +17,146 @@
 
-import "./INode.sol";
-import "../libraries/Cloneable.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
-contract Node is Cloneable, INode {
-    using SafeMath for uint256;
+struct NodeProps {
+    // Hash of the state of the chain as of this node
+    bytes32 stateHash;
 
-    /// @notice Hash of the state of the chain as of this node
-    bytes32 public override stateHash;
+    // Hash of the data that can be challenged
+    bytes32 challengeHash;
 
-    /// @notice Hash of the data that can be challenged
-    bytes32 public override challengeHash;
+    // Hash of the data that will be committed if this node is confirmed
+    bytes32 confirmData;
 
-    /// @notice Hash of the data that will be committed if this node is confirmed
-    bytes32 public override confirmData;
+    // Index of the node previous to this one
+    uint256 prev;
 
-    /// @notice Index of the node previous to this one
-    uint256 public override prev;
+    // Deadline at which this node can be confirmed
+    uint256 deadlineBlock;
 
-    /// @notice Deadline at which this node can be confirmed
-    uint256 public override deadlineBlock;
+    // Deadline at which a child of this node can be confirmed
+    uint256 noChildConfirmedBeforeBlock;
 
-    /// @notice Deadline at which a child of this node can be confirmed
-    uint256 public override noChildConfirmedBeforeBlock;
+    // Number of stakers staked on this node. This includes real stakers and zombies
+    uint256 stakerCount;
 
-    /// @notice Number of stakers staked on this node. This includes real stakers and zombies
-    uint256 public override stakerCount;
+    // Address of the rollup contract to which this node belongs
+    address rollup;
 
-    /// @notice Mapping of the stakers staked on this node with true if they are staked. This includes real stakers and zombies
-    mapping(address => bool) public override stakers;
+    // This value starts at zero and is set to a value when the first child is created. After that it is constant until the node is destroyed or the owner destroys pending nodes
+    uint256 firstChildBlock;
 
-    /// @notice Address of the rollup contract to which this node belongs
-    address public rollup;
+    // The number of the latest child of this node to be created
+    uint256 latestChildNumber;
+}
 
-    /// @notice This value starts at zero and is set to a value when the first child is created. After that it is constant until the node is destroyed or the owner destroys pending nodes
-    uint256 public override firstChildBlock;
+struct Node {
+    // Mapping of the stakers staked on this node with true if they are staked. This includes real stakers and zombies
+    mapping(address => bool) stakers;
+    
+    // All other Node data
+    NodeProps props;
+}
 
-    /// @notice The number of the latest child of this node to be created
-    uint256 public override latestChildNumber;
+/**
+ * @notice Utility functions for NodeProps
+ */
+library NodePropsLib {
```

Comment:
Any reason not to have these functions in the NodeLib? we already have the typing to be sure if a given param is node or nodeprop

Reply:
It's because the extend the NodeProps struct - the first arg of all these functions is NodeProps

Reply:
Merged now that we've removed stakers